tests/lib/bin/user-tool

```diff
@@ -1,42 +1,55 @@
-#!/bin/bash
+#!/usr/bin/env any-python
+from __future__ import print_function, absolute_import, unicode_literals
 
-set -e
+import argparse
+import os
+import subprocess
 
-# remove_user_with_group remove the given username from the system and also
-# cleans the group
-remove_user_with_group() {
-    REMOVE_USER="$1"
+# Define MYPY as False and use it as a conditional for typing import. Despite
+# this declaration mypy will really treat MYPY as True when type-checking.
+# This is required so that we can import typing on Python 2.x without the
+# typing module installed. For more details see:
+# https://mypy.readthedocs.io/en/latest/common_issues.html#import-cycles
+MYPY = False
+if MYPY:
+    from typing import Text
 
-    if [ -z "$REMOVE_USER" ]; then
-        echo "please provide a username to remove"
-        return 1
-    fi
 
-    if [ -e /var/lib/extrausers/passwd ]; then
-        userdel --extrausers --force "${REMOVE_USER}" || true
-    else
-        userdel --force --remove "${REMOVE_USER}" || true
-        # some systems do not set "USERGROUPS_ENAB yes" so we need to cleanup
+def remove_user_with_group(user_name):
+    # type: (Text) -> None
+    """remove the user and group with the same name, if present."""
+    if os.path.exists("/var/lib/extrausers/passwd"):
+        subprocess.call(["userdel", "--extrausers", "--force", user_name])
```

Comment:
I think as a followup this should also use "--remove" here. I know the original shell did not but that is a bug  in this silly shell.

Reply:
I'll make that so.

Reply:
This is now https://github.com/snapcore/snapd/pull/7328